plugins/inputs/netdata/netdata.go

```diff
@@ -0,0 +1,291 @@
+	server.currGatherTime = time.Now().Unix()
+
+	// first time gather data from the past 10s
+	if server.lastGatherTime == 0 {
+		server.lastGatherTime = time.Now().Unix() - 10
+	}
+
+	chartRequestURL, err := url.Parse(server.Url + chartsURL)
+	if err != nil {
+		acc.AddError(Errorf(err, "error parsing charts url : %s", server.Url+chartsURL))
+		return
+	}
+
+	chartRespBody, err := n.sendRequest(chartRequestURL, server)
+	if err != nil {
+		acc.AddError(Errorf(err, "error getting netdata charts, url : %s", chartRequestURL.String()))
+		return
+	}
+
+	var chartResp ResponseCharts
+	if err := json.Unmarshal([]byte(chartRespBody), &chartResp); err != nil {
+		acc.AddError(Errorf(err, "error unmarshalling netdata charts"))
+		return
+	}
+
+	for _, chart := range chartResp.Charts {
+		dataStringURL := n.buildChartDataURL(server, chart)
+
+		dataRequestURL, err := url.Parse(dataStringURL)
+
+		if err != nil {
+			acc.AddError(Errorf(err, "error parsing chart data url : %s", dataStringURL))
+			continue
+		}
+
+		dataRespBody, err := n.sendRequest(dataRequestURL, server)
+		if err != nil {
+			acc.AddError(Errorf(err, "error getting netdata chart data, url : %s", dataRequestURL.String()))
+			continue
+		}
+
+		var dataResp ResponseData
+		if err := json.Unmarshal([]byte(dataRespBody), &dataResp); err != nil {
+			acc.AddError(Errorf(err, "error unmarshalling netdata chart data"))
+			continue
+		}
+
+		dimensions := strings.Join(dataResp.DimensionNames, ",")
+
+		var measurement string
+		if n.PrefixChartIDByHostname {
+			measurement = "netdata." + chartResp.Hostname + "." + chart.ID
+		} else {
+			measurement = "netdata." + chart.ID
+		}
+
+		for _, chartData := range dataResp.Result.Data {
+			fields := make(map[string]interface{})
+			time := time.Unix(int64(chartData[0].(float64)), 0)
+			values := chartData[1:]
+			for i, dimension := range dataResp.DimensionNames {
+				if values[i] != nil {
+					fields[dimension] = values[i]
+				}
+			}
+
+			tags := map[string]string{
+				"server":     chartRequestURL.Host,
+				"hostname":   chartResp.Hostname,
+				"type":       chart.Type,
+				"family":     chart.Family,
+				"dimensions": dimensions,
+			}
+
+			acc.AddFields(measurement, fields, tags, time)
+		}
+	}
+	server.lastGatherTime = server.currGatherTime
+}
+
+func (n *Netdata) buildChartDataURL(s *Server, c Chart) string {
+	return s.Url +
+		c.DataURL +
+		"&group=" + n.Group +
+		"&options=absolute|jsonwrap|seconds&after=" +
+		strconv.FormatInt(s.lastGatherTime, 10) +
+		"&before=" +
+		strconv.FormatInt(s.currGatherTime, 10) +
+		"&points=" +
+		strconv.Itoa(n.Points)
+}
+
+func (n *Netdata) sendRequest(url *url.URL, server *Server) (string, error) {
+	headers := map[string]string{
+		"Content-Type": "application/json",
+		"Accept":       "application/json",
+	}
+	method := "GET"
+	content := bytes.NewBufferString("")
+	if server.BasicAuthPassword != "" && server.BasicAuthUsername != "" {
```

Comment:
I guess username without or with empty password is forbidden!? Otherwise it will break here.